### PR TITLE
feat(ethash/proofs): generate proofs using ethash-relayer.

### PR DESCRIPTION
**Pull request by shekohex**
Context: [Notion](https://www.notion.so/Implement-ETH-light-client-service-worker-9e881c5f1d1040e3b8d2f14acad925f5).

This PR is the next step after #6.
This adds support for generating Proofs inside the off-chain worker.

Next: testing this out!

related issues: #14, #15

### CODE REVIEW

**Review by drewstone — APPROVED** [review]
Looks great! Have you tried running a local node to see how it works?

**Comment by shekohex** [comment]
Not yet, I will do it!

**Comment by shekohex** [comment]
I removed the old code, and now we are using external relayer: https://github.com/edgeware-builders/ethash-relyer

File: pallets/worker/src/lib.rs [hunk 1]
```diff
@@ -372,128 +374,41 @@ decl_module! {
...
+                        header_queue.push_front(header);
+                        None
+                    }
                 } else {
                     debug::native::info!("Initializing with header #: {:?}!", header.number);
-                    let dags_start_epoch: u64 = header.number / 30000;
+                    let dags_start_epoch = header.number.as_u64() / 30000;
                     Some(Call::init(
                         dags_start_epoch,
                         vec![],
```
Comment (drewstone) [inline pallets/worker/src/lib.rs:414]: Maybe add roots here too?

Comment (shekohex) [reply pallets/worker/src/lib.rs:427]: yeah, I guess we could add the roots as hardcoded static array in `constatnts` module.

**Comment by drewstone** [comment]
Yea, I think its LE. You can also browse this repo and see if it already exists: https://github.com/paritytech/parity-common/blob/master/ethereum-types/src/uint.rs

**Comment by shekohex** [comment]
> Yea, I think its LE...

I tried running it now with LE instead, it does not work, INFURA returns `null` for that block, so it seems that it must be in BE.

**Pull request merged**